### ISSUE TITLE
KerasClassifier.predict can't handle one dimensional labels

### ISSUE THREAD

**Issue by Chaoste** [keras-team/keras#11818]
Keras==2.2.4

Hi guys,

Setup:
I'm currently working with the sklearn implementation of a MLPClassifier. Since I want to have one interface for all further (custom) models I decided to start from your interface KerasClassifier. So I created my own class which extends KerasClassifier but `__call__` returns an instance of sklearn.neural_network.MLPClassifier. I use this model for a multi-classification task with three classes [-1, 0, 1]. The training works fine calling `super().__fit__(X, y, **kwargs)`.

Issue:
The predict method of KerasClassifier has a strange behaviour. My model returns its predictions with the shape `(n_samples,)`. Having only one dimension the predict method only returns one number (-1 in my case) and I was wondering why. I found the reason to be in [these lines](https://github.com/keras-team/keras/blob/master/keras/wrappers/scikit_learn.py#L230):
```
proba = self.model.predict(x, **kwargs)
if proba.shape[-1] > 1:
    classes = proba.argmax(axis=-1)
else:
    classes = (proba > 0.5).astype('int32')
return self.classes_[classes]
```
This method doesn't consider to get labels with the shape (n_samples,). Since this is called a scikit_learn wrapper I suggest to add this case to the predict method. The fitting method already handles my format by transforming the classes [-1, 0, 1] into [0, 1, 2]. Now it needs to call `return self.classes_[classes]` and it would work fine. An if case for `proba` having one dimension and `self.classes_` being greater than two would be sufficient and shouldn't break any previous functionality.

Greetings,
Thomas

**Comment by Chaoste** [comment]
Since my issue might be written a bit confusing, here my suggestion for the predict method instead of the above code snippet:

```
    proba = self.model.predict(x, **kwargs)
    if len(self.classes_) > 2: and len(proba.shape) == 1:
        classes = proba
    if proba.shape[-1] > 1:
        classes = proba.argmax(axis=-1)
    else:
        classes = (proba > 0.5).astype('int32')
    return self.classes_[classes]
```

**Comment by Chaoste** [comment]
:'(
